src/app/src/components/AnimatedArcsAndMap/useCreateArcPath.tsx

```diff
@@ -0,0 +1,128 @@
+import { useCallback } from 'react';
+import { useMap } from 'react-leaflet';
+import L, { LatLngLiteral, LatLngTuple } from 'leaflet';
+import { useGetSpendingOverTimeQuery } from '../../api';
+import { MonthlySpendingOverTime } from '../../types/api';
+import { PROGRESS_FINAL_STEP } from '../../util';
+import { DC_CENTER, MONTHLY_TIME_DURATION } from '../../constants';
+import { StateFeature } from '../states.geojson';
+
+import polylabel from 'polylabel';
+
+export default function useCreateArcPath(
+    arcPathsReference: React.MutableRefObject<Record<string, L.Curve>>
+) {
+    const { data: spending } = useGetSpendingOverTimeQuery();
```

Comment:
I started implementing, but I wonder if it might be better to leave it this way and then just better consolidate the null checking into one wrapping if statement. This hook uses the `spending` data to find when to start the path animation in the timeline, which it does by checking for the first non-zero `aggregated_amount` value in a state's results array. We only really need to do that once per state at the initial render/`onEachFeature` call, so we might end up redundantly creating the callback fn by adding `spendingAtTimeByState` as a hook dependency—although there's probably already redundancy having `arcPathsReference` as a dependency so maybe the readability is worth it. What do you think?